packit_service/worker/events/koji.py

```diff
@@ -6,42 +6,209 @@
+        return self._build_model
+
+    @property
+    def db_trigger(self) -> Optional[AbstractTriggerDbType]:
+        if not self._db_trigger and self.build_model:
+            self._db_trigger = self.build_model.get_trigger_object()
+        return self._db_trigger
+
+    @property
+    def target(self) -> Optional[str]:
+        if not self._target and self.build_model:
+            self._target = self.build_model.target
+        return self._target
+
+    @staticmethod
+    def get_koji_build_logs_url(
+        rpm_build_task_id: int,
+        koji_logs_url: str = "https://kojipkgs.fedoraproject.org",
+    ) -> str:
+        """
+        Constructs the log URL for the given Koji task.
+        You can redefine the Koji instance using the one defined in the service config.
+        """
+        return (
+            f"{koji_logs_url}//work/tasks/"
+            f"{rpm_build_task_id % 10000}/{rpm_build_task_id}/build.log"
+        )
+
+    @staticmethod
+    def get_koji_rpm_build_web_url(
+        rpm_build_task_id: int,
+        koji_web_url: str = "https://koji.fedoraproject.org",
+    ) -> str:
+        """
+        Constructs the web URL for the given Koji task.
+        You can redefine the Koji instance using the one defined in the service config.
+        """
+        return f"{koji_web_url}/koji/taskinfo?taskID={rpm_build_task_id}"
+
+    def get_dict(self, default_dict: Optional[Dict] = None) -> dict:
+        result = super().get_dict()
+        result.pop("_build_model")
+        result.pop("_build_model_searched")
+        result.pop("_db_trigger")
+        return result
+
+
+class KojiBuildEvent(AbstractKojiEvent):
     def __init__(
         self,
         build_id: int,
         state: KojiBuildState,
+        package_name: str,
+        branch_name: str,
+        commit_sha: str,
+        namespace: str,
+        repo_name: str,
+        project_url: str,
+        epoch: str,
+        version: str,
+        release: str,
+        rpm_build_task_id: int,
+        web_url: Optional[str] = None,
         old_state: Optional[KojiBuildState] = None,
+    ):
+        super().__init__(build_id)
+        self.state = state
+        self.old_state = old_state
+        self.package_name = package_name
+        self.rpm_build_task_id = rpm_build_task_id
+        self.epoch = epoch
+        self.version = version
+        self.release = release
+        self.web_url = web_url
+
+        self.branch_name = branch_name
+        self._commit_sha = commit_sha  # we know  it, no need to get it from db
+        self.namespace = namespace
+        self.repo_name = repo_name
+        self.project_url = project_url
+
+    @property
+    def commit_sha(self) -> Optional[str]:  # type:ignore
+        # mypy does not like properties
+        return self._commit_sha
+
+    @property
+    def build_model(self) -> Optional[KojiBuildModel]:
+        if not super().build_model:
+            self._build_model = KojiBuildModel.create(
```

Comment:
would it make sense to have a new boolean field in the model such as scratch or production so that we can differentiate between scratch and non-scratch builds since e.g. status differs for them?

Reply:
That's a good point. I'll send a follow-up if it is ok -- I have a few other small improvements in mind around this.